### PR TITLE
Detect spring boot dependency version

### LINKED ISSUE #5: Support SpringBoot 2.0 as an additional choice (vs. SpringBoot 1.5)
* Key off of parent starter POM version
* generate different Liberty server.xml features.

### LINKED ISSUE #9: Enhance "first" integration test to actually start server and check endpoint somehow.
Currently it just verifies that we can do a mvn build successfully.

### PR DESCRIPTION
**Pull request by cherylking**
Fixes #5  and part of #12

### CODE REVIEW

File: src/main/java/boost/project/PackageSpringBootAppWithLibertyMojo.java [hunk 1]
```diff
@@ -223,4 +253,19 @@ private void createUberJar() throws MojoExecutionException {
...
 	}
+
+	/**
+	* Detect spring boot version dependency
+	*/
+	private String findSpringBootVersion(MavenProject project) {
+        	String version = null;
+        	List<Dependency> dependencies = project.getDependencies();
+        	for(Dependency dep: dependencies) {
+            		if("org.springframework.boot".equals(dep.getGroupId())){
```
Comment (scottkurz) [inline src/main/java/boost/project/PackageSpringBootAppWithLibertyMojo.java]: Would it be better to narrow this down to the "spring-boot-starter-parent" artifact specifically within the group?   Not sure if there's a valid case where the parent starter comes from one version but some other dependency in the group comes from another (1.5.x vs. 2.0.x).

Comment (cherylking) [reply src/main/java/boost/project/PackageSpringBootAppWithLibertyMojo.java]: This is how Anjum recommended checking the version of the spring boot dependency. The dependency versions end up coming from the parent pom (several levels of indirection up). Plus, the application could reference a different parent pom that then lists the spring boot one as its parent. In which case, the version would not be the one we are looking for.

Comment (cherylking) [reply src/main/java/boost/project/PackageSpringBootAppWithLibertyMojo.java]: Let's talk more tomorrow about this mix version case and whether it is valid for Open Liberty.

**Comment by cherylking** [comment]
@scottkurz Ready for another look.

**Comment by scottkurz** [comment]
Reviewed.  Added https://github.com/OpenLiberty/boost/issues/21 for follow-up integration testing.

**Pull request merged**